commerce-api/src/main/java/com/liferay/commerce/payment/engine/CommercePaymentEngine.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.payment.engine;
+
+import aQute.bnd.annotation.ProviderType;
+
+import com.liferay.commerce.payment.method.CommercePaymentMethod;
+import com.liferay.commerce.payment.result.CommercePaymentResult;
+import com.liferay.portal.kernel.exception.PortalException;
+
+import java.util.List;
+import java.util.Locale;
+
+import javax.servlet.ServletRequest;
+
+/**
+ * @author Luca Pellizzon
+ */
+@ProviderType
+public interface CommercePaymentEngine {
+
+	public CommercePaymentResult cancelPayment(
+			long commerceOrderId, String transactionId,
+			ServletRequest httpServletRequest)
```

Comment:
SourceFormatter was complaining about using `HttpServletRequest` in an interface.
The original code was using `HttpServletRequest` directly and I think it was better, but SF hates me 😃

Reply:
I see :). I think it would complain about `ServletRequest`, too, it just isn't common enough for them to add in that rule :). I'll switch them to `HttpServletRequest` and add one ignore to the `source-formatter-suppressions.xml`.